python/mergeTreeClustering.py

```diff
@@ -0,0 +1,105 @@
+isabelvti.TimeArray = 'None'
+
+all_JT_group = []
+all_ST_group = []
+
+for scalarField in isabelvti.PointArrayStatus:
+  # create a new 'TTK Merge and Contour Tree (FTM)'
+  tTKMergeandContourTreeFTM1 = TTKMergeandContourTreeFTM(Input=isabelvti)
+  tTKMergeandContourTreeFTM1.ScalarField = ['POINTS', scalarField]
+  tTKMergeandContourTreeFTM1.InputOffsetField = ['POINTS', scalarField]
+  tTKMergeandContourTreeFTM1.TreeType = 'Join Tree'
+  
+  # create a new 'Group Datasets'
+  groupDatasets1 = GroupDatasets(Input=[tTKMergeandContourTreeFTM1, OutputPort(tTKMergeandContourTreeFTM1,1), OutputPort(tTKMergeandContourTreeFTM1,2)])
+  
+  all_JT_group.append(groupDatasets1)
+  
+  # create a new 'TTK Merge and Contour Tree (FTM)'
+  tTKMergeandContourTreeFTM2 = TTKMergeandContourTreeFTM(Input=isabelvti)
+  tTKMergeandContourTreeFTM2.ScalarField = ['POINTS', scalarField]
+  tTKMergeandContourTreeFTM2.InputOffsetField = ['POINTS', scalarField]
+  tTKMergeandContourTreeFTM2.TreeType = 'Split Tree'
+  
+  # create a new 'Group Datasets'
+  groupDatasets2 = GroupDatasets(Input=[tTKMergeandContourTreeFTM2, OutputPort(tTKMergeandContourTreeFTM2,1), OutputPort(tTKMergeandContourTreeFTM2,2)])
+  
+  all_ST_group.append(groupDatasets2)
+
+# create a new 'Group Datasets'
+mt_JT_all = GroupDatasets(Input=all_JT_group)
+
+# create a new 'Group Datasets'
+mT_all = GroupDatasets(Input=all_ST_group)
+
+# create a new 'TTK MergeTreeClustering'
+tTKMergeTreeClustering1 = TTKMergeTreeClustering(Input=mT_all,
+    OptionalInputclustering=mt_JT_all)
+tTKMergeTreeClustering1.ComputeBarycenter = 1
+tTKMergeTreeClustering1.NumberOfClusters = 3
+tTKMergeTreeClustering1.Deterministic = 1
+tTKMergeTreeClustering1.DimensionSpacing = 0.1
+tTKMergeTreeClustering1.PersistenceThreshold = 2.0
+tTKMergeTreeClustering1.ImportantPairs = 34.0
+tTKMergeTreeClustering1.MaximumNumberofImportantPairs = 3
+tTKMergeTreeClustering1.MinimumNumberofImportantPairs = 2
+tTKMergeTreeClustering1.ImportantPairsSpacing = 15.0
+tTKMergeTreeClustering1.NonImportantPairsProximity = 0.15
+
+# create a new 'Group Datasets'
+groupDatasets14 = GroupDatasets(Input=[tTKMergeTreeClustering1, OutputPort(tTKMergeTreeClustering1,1)])
+
+# create a new 'TTK FlattenMultiBlock'
+tTKFlattenMultiBlock2 = TTKFlattenMultiBlock(Input=groupDatasets14)
+
+# create a new 'TTK MergeTreeDistanceMatrix'
+tTKMergeTreeDistanceMatrix2 = TTKMergeTreeDistanceMatrix(Input=tTKFlattenMultiBlock2)
+tTKMergeTreeDistanceMatrix2.PersistenceThreshold = 2.0
+
+# create a new 'TTK DimensionReduction'
+tTKDimensionReduction2 = TTKDimensionReduction(Input=tTKMergeTreeDistanceMatrix2,
+    ModulePath='default')
+tTKDimensionReduction2.InputColumns = ['Tree00', 'Tree01', 'Tree02', 'Tree03', 'Tree04', 'Tree05', 'Tree06', 'Tree07', 'Tree08', 'Tree09', 'Tree10', 'Tree11', 'Tree12', 'Tree13', 'Tree14']
+tTKDimensionReduction2.InputIsaDistanceMatrix = 1
+tTKDimensionReduction2.UseAllCores = 0
```

Comment:
Do you really run into issues otherwise?
Are you using MDS? (default method)

Reply:
Yes the script uses MDS, and indeed the script does not work using parallelism.

Reply:
can you please add a comment "MDS is unstable in parallel mode".